ui/app/components/app/connected-accounts-list/connected-accounts-list.component.js

```diff
@@ -26,31 +23,38 @@ export default class ConnectedAccountsList extends PureComponent {
       name: PropTypes.string.isRequired,
       lastActive: PropTypes.number,
     })).isRequired,
-    permissions: PropTypes.arrayOf(PropTypes.shape({
-      key: PropTypes.string.isRequired,
-    })),
+    connectAccount: PropTypes.func.isRequired,
     selectedAddress: PropTypes.string.isRequired,
-    addPermittedAccount: PropTypes.func.isRequired,
-    removePermittedAccount: PropTypes.func.isRequired,
+    removePermittedAccount: PropTypes.func,
     setSelectedAddress: PropTypes.func.isRequired,
+    shouldRenderListOptions: (props, propName, componentName) => {
+      if (props[propName]) {
```

Comment:
This wouldn't apply if `shouldRenderListOptions` was `false`

```suggestion
      if (props[propName] !== undefined) {
```

Reply:
True! Since `removePermittedAccount` is not required if `shouldRenderListOptions` is `false`, it required some more changes. See c78e377